src/http/client/client.cr

```diff
@@ -485,6 +511,10 @@ class HTTP::Client
     end
   end
 
+  private def self.allowed_scheme?(uri : URI)
+    uri.scheme =~ /https?/
```

Comment:
But `starts_with?("http")` is not correct, what if the scheme is "httpfoo"? We need to check it's exactly "http" or "https".

We can make name the method that checks the scheme something like "ssl_flag":

```crystal
protected def self.ssl_flag(uri)
  scheme = uri.scheme
  case scheme
  when nil
    raise "missing scheme: #{uri}"
  when "http"
    false
  when "https"
    true
  else
    raise "Unsupported scheme: #{scheme}"
  end
end
```

So the usage is just:

```crystal
ssl = self.ssl_flag(uri)
```

and the checks and raise are in a single place. We can use this in `self.exec` too.